Products/GenericSetup/tool.py

```diff
@@ -1166,7 +1178,16 @@ def _runImportStepsFromContext(self,
 
         detect_steps = steps is None
 
-        for profile_id in chain:
+        last_index = len(chain) - 1
+        for index, profile_id in enumerate(chain):
+            if not always_apply_profiles and index != last_index:
```

Comment:
Surely `always_apply_profiles` should be outside the loop? 

If chain has a length then for `profile_id in chain[:-1]:` should work too, shouldn't it?